exporters/elasticsearch/src/es_log_exporter.cc

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "opentelemetry/exporters/elasticsearch/es_log_exporter.h"
+#include "opentelemetry/exporters/elasticsearch/es_log_recordable.h"
+
+namespace nostd       = opentelemetry::nostd;
+namespace sdklogs     = opentelemetry::sdk::logs;
+namespace http_client = opentelemetry::ext::http::client;
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace logs
+{
+/**
+ * This class handles the response message from the Elasticsearch request
+ */
+class ResponseHandler : public http_client::EventHandler
+{
+public:
+  /**
+   * Automatically called when the response is received, store the body into a string and notify any
+   * threads blocked on this result
+   */
+  void OnResponse(http_client::Response &response) noexcept override
+  {
+    // Store the body of the request
+    body_ = std::string(response.GetBody().begin(), response.GetBody().end());
+
+    // Set the response_received_ flag to true and notify any threads waiting on this result
+    response_received_ = true;
+    cv_.notify_all();
+  }
+
+  /**
+   * A method the user calls to block their thread until the response is received, or the timeout is
+   * exceeded.
+   */
+  bool waitForResponse(unsigned int timeoutSec = 1)
+  {
+    std::mutex mutex_;
+    std::unique_lock<std::mutex> lk(mutex_);
+    cv_.wait_for(lk, std::chrono::milliseconds(1000 * timeoutSec));
+    return response_received_;
+  }
+
+  /**
+   * Returns the body of the response
+   */
+  std::string GetResponseBody()
+  {
+    if (!response_received_)
```

Comment:
nit: maybe check `if response_received` before entering this function instead?

Reply:
yea it's being checked in the export method already, I had this as a just in case kinda thing tho. I'll take it out tho since it's never gonna get called